classes/genimage.bbclass

```diff
@@ -92,19 +92,18 @@ do_genimage[depends] += "${@'${GENIMAGE_ROOTFS_IMAGE}:do_image_complete' if '${G
 GENIMAGE_TMPDIR  = "${WORKDIR}/genimage-tmp"
 GENIMAGE_ROOTDIR  = "${WORKDIR}/root"
 
-do_genimage[cleandirs] = "${GENIMAGE_TMPDIR} ${GENIMAGE_ROOTDIR}"
+do_genimage[cleandirs] = "${GENIMAGE_TMPDIR} ${GENIMAGE_ROOTDIR} ${B}"
 
 do_configure () {
     if grep -vq "@IMAGE@" ${WORKDIR}/genimage.config; then
         bbnote "genimage.config does not contain @IMAGE@ marker"
     fi
-
-    cp ${WORKDIR}/genimage.config ${B}/.config
 }
 
 do_genimage[dirs] = "${B}"
```

Comment:
This is not required anymore then as `[cleandirs]` already implies `[dirs]`

Reply:
Yes, in terms of ensuring those directories exist before the task is started (with the additional property of ensuring they are empty). But I don't think it also has the side effect of making the working directory for the task be the last element in cleandirs when dirs is absent. At least that's not how I read the documentation or `bitbake/lib/bb/build.py`.

Reply:
Hm, maybe meta-ptx master as-is doesn't need the cwd of do_genimage to be set explicitly to ${B}. But I do need that because I currently work around that using `include` in the main config file got broken when `${S}` and `${B}` were separated by copying all my config fragments to ${B} in a do_genimage_prepend. Until I can rely on `--includepath ${S}` being passed and drop that workaround, ${B} must be the cwd of do_genimage.

Reply:
Ok, the cwd would be an argument to keep it.